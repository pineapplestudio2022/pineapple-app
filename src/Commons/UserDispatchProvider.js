import React, {createContext} from 'react';
import {useReducer} from 'react';
import {cleanClientToken} from '../API/APIkit';

export const UserDispatch = createContext();
const initialState = {
  userId: '',
  email: '',
};

const reducer = (state, action) => {
  switch (action.type) {
    case 'SIGN_IN':
      return {
        userId: action.userId,
        email: action.email,
      };
    case 'SIGN_OUT':
      cleanClientToken();
      return {
        userId: '',
        email: '',
      };
    default:
      return state;
  }
};

const UserDispatchProvider = props => {
  const [state, dispatch] = useReducer(reducer, initialState);
  return (
    <UserDispatch.Provider
      value={{userId: state.userId, email: state.email, dispatch}}>
      {props.children}
    </UserDispatch.Provider>
  );
};

export default UserDispatchProvider;
