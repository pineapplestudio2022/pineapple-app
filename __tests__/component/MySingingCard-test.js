import React from 'react';
import {NativeBaseProvider} from 'native-base';
import {render} from '@testing-library/react-native';
import MySingingCard from '../../src/Components/MySingingCardComponent';
import ShallowRenderer from 'react-test-renderer/shallow';
import UserDispatchProvider from '../../src/Commons/UserDispatchProvider';

const wrapper = ({children}) => (
  <UserDispatchProvider>
    <NativeBaseProvider
      initialWindowMetrics={{
        frame: {x: 0, y: 0, width: 0, height: 0},
        insets: {top: 0, left: 0, right: 0, bottom: 0},
      }}>
      {children}
    </NativeBaseProvider>
  </UserDispatchProvider>
);

// let realUseContext;
// let useContextMock;
// // Setup mock
// beforeEach(() => {
//   realUseContext = React.useContext;
//   useContextMock = React.useContext = jest.fn();
// });
// // Cleanup mock
// afterEach(() => {
//   React.useContext = realUseContext;
// });

// test('mock hook', () => {
//   useContextMock.mockReturnValue('Test Value');
//   const element = new ShallowRenderer().render(<MySingingCard />);
//   expect(element).toMatchSnapshot(); //기존 스냅샷과 일치하는지
//   expect(element).toBeTruthy(); //컴포넌트가 null, undefinded 같이 falsy 한 값을 가지지 않았는지
// });

let props;
let component;

const getTempComponent = props => {
  return <MySingingCard {...props} />;
};

describe('[MySingingCard] Test', () => {
  //random.math mock
  const mockMath = Object.create(global.Math);
  mockMath.random = () => 0.5;
  global.Math = mockMath;

  props = {
    navigation: jest.fn(),
    id: '275',
    originalWorkId: '35',
    title: 'title_',
    genre: 'genre_',
    detail: 'detail_',
    setRefresh: jest.fn(),
  };

  component = getTempComponent(props);

  test('MySingingCard snapshot', () => {
    jest.useFakeTimers();
    const rendered = render(component, {wrapper});

    expect(rendered).toMatchSnapshot(); //기존 스냅샷과 일치하는지
    expect(rendered).toBeTruthy(); //컴포넌트가 null, undefinded 같이 falsy 한 값을 가지지 않았는지
  });
});
