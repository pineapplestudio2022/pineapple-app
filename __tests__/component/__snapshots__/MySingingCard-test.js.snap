// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[MySingingCard] Test MySingingCard snapshot 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    backgroundColor="#fafafa"
    height={164.5741399762752}
    rounded={8}
    style={
      Array [
        Object {
          "backgroundColor": "#fafafa",
          "borderRadius": 8,
          "height": 164.5741399762752,
          "width": 615.3846153846154,
        },
        Object {
          "shadowColor": "#8799a45b",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 1,
          "shadowRadius": 10,
        },
      ]
    }
    width={615.3846153846154}
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <View
        flexDirection="row"
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
          ]
        }
      >
        <View
          height={150.33214709371293}
          m={1}
          overflow="hidden"
          rounded={4}
          style={
            Array [
              Object {
                "borderRadius": 4,
                "height": 150.33214709371293,
                "marginBottom": 4,
                "marginLeft": 4,
                "marginRight": 4,
                "marginTop": 4,
                "overflow": "hidden",
                "width": 182.69230769230768,
              },
            ]
          }
          width={182.69230769230768}
        >
          <Image
            accessibilityLabel=" "
            alt=" "
            height="100%"
            maxWidth="100%"
            onError={[Function]}
            resizeMode="cover"
            source={
              Object {
                "testUri": "../../../src/Assets/Image/Top_music/top_music_6.jpg",
              }
            }
            style={
              Array [
                Object {
                  "height": "100%",
                  "maxWidth": "100%",
                  "width": "100%",
                },
              ]
            }
            width="100%"
          />
        </View>
        <View
          style={
            Array [
              Object {
                "width": 16,
              },
            ]
          }
          width={4}
        />
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <View
            flexDirection="row"
            style={
              Array [
                Object {
                  "flexDirection": "row",
                },
              ]
            }
          >
            <Text
              color="#1a1b1c"
              fontFamily="body"
              fontSize={33.80850587417971}
              fontStyle="normal"
              fontWeight={600}
              py={1}
              style={
                Array [
                  Object {
                    "color": "#1a1b1c",
                    "fontSize": 33.80850587417971,
                    "fontStyle": "normal",
                    "fontWeight": "600",
                    "paddingBottom": 4,
                    "paddingTop": 4,
                  },
                ]
              }
            >
              title_
            </Text>
            <View
              style={
                Array [
                  Object {},
                ]
              }
            />
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "height": 25.319098457888494,
                  "opacity": 1,
                  "position": "absolute",
                  "right": 0,
                  "top": 8,
                  "width": 30.769230769230763,
                }
              }
            >
              <Image
                accessibilityLabel=" "
                alt=" "
                height="100%"
                maxWidth="100%"
                onError={[Function]}
                resizeMode="contain"
                source={
                  Object {
                    "testUri": "../../../src/Assets/Image/challenge/icon_challenge_trash.png",
                  }
                }
                style={
                  Array [
                    Object {
                      "height": "100%",
                      "maxWidth": "100%",
                      "width": "100%",
                    },
                  ]
                }
                width="100%"
              />
            </View>
          </View>
          <View
            style={
              Array [
                Object {},
              ]
            }
          />
          <Text
            color="#858c92"
            fontFamily="body"
            fontSize={21.876092036233935}
            fontStyle="normal"
            fontWeight={500}
            numberOfLines={1}
            style={
              Array [
                Object {
                  "color": "#858c92",
                  "fontSize": 21.876092036233935,
                  "fontStyle": "normal",
                  "fontWeight": "500",
                },
                Object {
                  "height": 28.483985765124554,
                  "width": 326.9230769230769,
                },
              ]
            }
          >
            detail_
          </Text>
          <View
            style={
              Array [
                Object {},
              ]
            }
          />
          <View
            alignItems="center"
            flexDirection="row"
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                },
                Object {
                  "height": 26.901542111506526,
                  "width": 326.9230769230769,
                },
              ]
            }
          >
            <Text
              color="#000000"
              fontFamily="body"
              fontSize={21.876092036233935}
              fontStyle="normal"
              fontWeight={500}
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontSize": 21.876092036233935,
                    "fontStyle": "normal",
                    "fontWeight": "500",
                  },
                ]
              }
            >
              장르 :
            </Text>
            <View
              style={
                Array [
                  Object {
                    "width": 4,
                  },
                ]
              }
              width={1}
            />
            <Text
              color="#858c92"
              fontFamily="body"
              fontSize={21.876092036233935}
              fontStyle="normal"
              fontWeight={500}
              style={
                Array [
                  Object {
                    "color": "#858c92",
                    "fontSize": 21.876092036233935,
                    "fontStyle": "normal",
                    "fontWeight": "500",
                  },
                ]
              }
            >
              genre_
            </Text>
          </View>
          <View
            style={
              Array [
                Object {},
              ]
            }
          />
          <View
            flexDirection="row"
            mt={1}
            style={
              Array [
                Object {
                  "flexDirection": "row",
                  "marginTop": 4,
                },
              ]
            }
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "backgroundColor": "#0fefbd",
                  "borderRadius": 4,
                  "height": 37.97864768683274,
                  "justifyContent": "center",
                  "opacity": 1,
                  "width": 146.1538461538462,
                }
              }
            >
              <View
                flexDirection="row"
                style={
                  Array [
                    Object {
                      "flexDirection": "row",
                    },
                  ]
                }
              >
                <Image
                  accessibilityLabel=" "
                  alt=" "
                  maxWidth="100%"
                  onError={[Function]}
                  resizeMode="contain"
                  source={
                    Object {
                      "testUri": "../../../src/Assets/Image/challenge/icon_challenge_playmusic2.png",
                    }
                  }
                  style={
                    Array [
                      Object {
                        "maxWidth": "100%",
                      },
                      Object {
                        "marginLeft": 7,
                        "width": 30.769230769230763,
                      },
                    ]
                  }
                />
                <View
                  style={
                    Array [
                      Object {},
                    ]
                  }
                />
                <Text
                  color="#fafafa"
                  fontFamily="body"
                  fontSize={23.864827675891565}
                  fontStyle="normal"
                  fontWeight="bold"
                  marginLeft={3}
                  style={
                    Array [
                      Object {
                        "color": "#fafafa",
                        "fontSize": 23.864827675891565,
                        "fontStyle": "normal",
                        "fontWeight": "700",
                        "marginLeft": 12,
                      },
                    ]
                  }
                >
                  재 생
                </Text>
              </View>
            </View>
            <View
              style={
                Array [
                  Object {
                    "width": 36,
                  },
                ]
              }
              width={9}
            />
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "backgroundColor": "#0fefbd",
                  "borderRadius": 4,
                  "height": 37.97864768683274,
                  "justifyContent": "center",
                  "opacity": 1,
                  "width": 146.1538461538462,
                }
              }
            >
              <View
                flexDirection="row"
                style={
                  Array [
                    Object {
                      "flexDirection": "row",
                    },
                  ]
                }
              >
                <Image
                  accessibilityLabel=" "
                  alt=" "
                  maxWidth="100%"
                  onError={[Function]}
                  resizeMode="contain"
                  source={
                    Object {
                      "testUri": "../../../src/Assets/Image/challenge/icon_challenge_mic.png",
                    }
                  }
                  style={
                    Array [
                      Object {
                        "maxWidth": "100%",
                      },
                      Object {
                        "marginLeft": 7,
                        "width": 30.769230769230763,
                      },
                    ]
                  }
                />
                <View
                  style={
                    Array [
                      Object {},
                    ]
                  }
                />
                <Text
                  color="#fafafa"
                  fontFamily="body"
                  fontSize={23.864827675891565}
                  fontStyle="normal"
                  fontWeight="bold"
                  marginLeft={1}
                  style={
                    Array [
                      Object {
                        "color": "#fafafa",
                        "fontSize": 23.864827675891565,
                        "fontStyle": "normal",
                        "fontWeight": "700",
                        "marginLeft": 4,
                      },
                    ]
                  }
                >
                  재 도 전
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</RNCSafeAreaProvider>
`;
